bin/verify_rke2_charts

```diff
@@ -0,0 +1,114 @@
+#!/bin/sh
+
+set -e
+
+. libstd-ecm.sh
+. librancher-ecm.sh
+
+# Defaults
+REPOSITORY=rancher/rke2
+LOCAL_RKE2=''
+BRANCH='master'
+IGNORE_LIST=''
+
+usage() {
+    echo "usage: ${0} [hbilr]
+    -b    set branch. Default: ${BRANCH}
+    -r    set Github repository name. Default: ${REPOSITORY}
+    -i    add white space separted chart ignore list. ie: 'rke2-metrics-server rancher-vsphere-cpi'
+    -l    use local repository path. -r is ignored when this is set.
+    -x    print debug messages
+    -h    show help
+
+examples:
+    ${0}
+    ${0} -i 'rke2-metrics-server rancher-vsphere-cpi' -r my-gh-user/rke2
+    ${0} -l ../rke2 -b release-1.22"
+}
+
+while getopts 'b:i:l:r:xh' c; do
+    case "${c}" in
+    b)
+        BRANCH=${OPTARG}
+        ;;
+    x)
+        set_debug
+        ;;
+    i)
+        IGNORE_LIST=${OPTARG}
+        ;;
+    l)
+        LOCAL_RKE2=${OPTARG}
+        ;;
+    r)
+        REPOSITORY=${OPTARG}
+        ;;
+    h)
+        usage
+        exit 0
+        ;;
+    *)
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+setup_tmp
+
+# gh environment variable overrides
+export GH_REPO="${REPOSITORY}"
+export PAGER='cat'
+
+# Get required file either locally or remotely
+if [ -z "${LOCAL_RKE2}" ]; then
+    cd "${TMP_DIR}"
+
+    has_gh
+
+    gh api -XGET --field "ref=${BRANCH}" --header "Accept:application/vnd.github.VERSION.raw" \
+        '/repos/{owner}/{repo}/contents/Dockerfile' >Dockerfile
+else
+    cd "${LOCAL_RKE2}"
+
+    if [ "$(git rev-parse --is-shallow-repository)" = 'false' ]; then
+        git checkout -q "${BRANCH}"
+    fi
+fi
+
+has_yq
```

Comment:
Can we move this up to just below the flag parse? Since `yq` is part of the critical hot path, I'd be good to fail fast on that not being available.

Reply:
I didn't think of that. Yeah, let me move it.